### PR TITLE
Cavebuster Missile: Making Caves less of a hell for POs and MARINES alike

### PR DESCRIPTION
**Pull request by JackTheJackhammer**
## About The Pull Request

Adds Penetrator Missile for CAS
Adds base code for more CAS launched, Cave enabled, ordinance

The penetrator missile costs 200 CAS points and has a explosive strength of (2, 4 7)
The missile can fire in caves.

## Why It's Good For The Game

Marines often have a hard time pushing caves even with multiple miners and full map control. This can help remedy that issue. Caves are just too hard to push. Even with 5 b18s and multiple SGs, caves are a pain to push (trust me, it happens more than you think). Part of this problem is caused by the lack of fire support by marines. At first, when caves were being implemented, this would have sounded like a good idea. However, now with the introduction of pools and cocooning, caves have become a spot of delays. Xenos can flip a marine victory to a xeno victory by spamming walls, turrets and other form of slow down defenses (all forms of defenses that could be countered by fire support and wait until they accumulate enough larva to counter attack. The basics of game design state that everything should have a counter. Within caves, spam has no counter. 

NOTE: Marines get 33% of the normal amount of points (CAS POINTS) when at orbit 1 and 66% (give or take) at orbit 5

TL;DR:

The lack of fire support and wall spam makes caves too hard to push. This should help make that less of a problem

## Changelog
:cl:
add: Cave enabled CAS ordinance base code
add: GBU-1172 'Penetrator'
/:cl:

### CODE REVIEW

File: code/game/objects/structures/dropship_ammo.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@
 	var/point_cost = 0 //how many points it costs to build this with the fabricator, set to 0 if unbuildable.
 	///Type of ammo
 	var/ammo_type
+	var/Penetrate_Cave = FALSE
```
Comment (BraveMole) [inline code/game/objects/structures/dropship_ammo.dm:31]: 1) autodoc (see the pins in dev gen)
2) penetrate_cave, not Penetrate_Cave
3) This could probably be in a flag with transferable_ammo, rather than a bool

File: code/modules/shuttle/cas_plane.dm [hunk 2]
```diff
@@ -308,13 +308,11 @@
 		return
 	if(!GLOB.cameranet.checkTurfVis(get_turf_pixel(target)))
 		return
-	if(!active_weapon)
-		to_chat(source, "<span class='warning'>No active weapon selected!</span>")
-		return
```
Comment (BraveMole) [inline code/modules/shuttle/cas_plane.dm:313]: ?

File: code/modules/shuttle/cas_plane.dm [hunk 3]
```diff
@@ -308,13 +308,11 @@
...
 	var/area/A = get_area(target)
 	if(A.ceiling >= CEILING_DEEP_UNDERGROUND)
-		to_chat(source, "<span class='warning'>That target is too deep underground!</span>")
-		return
+		if(!active_weapon.ammo_equipped.Penetrate_Cave == TRUE)
```
Comment (BraveMole) [inline code/modules/shuttle/cas_plane.dm:313]: 1) penetrate_cave is already a boolean, no need to do if (TRUE == TRUE)
2) you tested this? looks like the boolean math is wrong. And even if it's doing the correct thing, this is very much shit code, redo it please

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/cas_plane.dm (hunk 2 above)
Comment (JackTheJackhammer) [reply code/modules/shuttle/cas_plane.dm:313]: Oh, I must have accidentally deleted that, I'll put it back in

File: code/modules/shuttle/cas_plane.dm (hunk 3 above)
Comment (JackTheJackhammer) [reply code/modules/shuttle/cas_plane.dm:313]: If by shitcode, you mean I shouldn't have that as a  == TRUE. Yes, Ill go fix that. If not, please elaborate.

**Comment by Tennessee116** [comment]
> The basics of game design state that everything should have a counter.

TGMC has a game design document, or am I misinterpreting this?

**Comment by JackTheJackhammer** [comment]
What I mean by "basics of game design" is not the TGMC design doc. But really just how most games (mostly multiplayer) are balanced and the such. What I am citing is not the DOC but just commonly used techniques of balance.

**Comment by BraveMole** [comment]
> The lack of fire support and wall spam makes caves too hard to push. This should help make that less of a problem

I hope it's not TOO good. Caves should still be a bit harder to push for marines, cause xenos are really at a big disadvantage in open ground. Although i agree that xenos are probably too strong in caves right now

**Comment by neohenry** [comment]
CAS and Condor powercreep.

**Comment by Losenis** [comment]
>I hope it's not TOO good. Caves should still be a bit harder to push for marines, cause xenos are really at a big disadvantage in open ground. Although i agree that xenos are probably too strong in caves right now

Caves will be inherently harder to push because it's an area that only has drawbacks compared to the outside. LV is a meme so I won't bring it as example but on big red you have to park the tadpole extremely far from whatever is in caves to push, and that's assuming you can pinpoint the location. Without a PC you're just boned in any attempt to widen gaps, whereas normal walls can be brought down even with bullets if ammo allows.

File: code/modules/shuttle/cas_plane.dm [hunk 4]
```diff
@@ -308,13 +308,14 @@
 		return
 	if(!GLOB.cameranet.checkTurfVis(get_turf_pixel(target)))
 		return
+	var/area/A = get_area(target)
```
Comment (BraveMole) [inline code/modules/shuttle/cas_plane.dm:311]: should be line 314 like it was

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Walarks** [comment]
> The basics of game design state that everything should have a counter.

Not everything needs to have a direct counter. I feel like this change would make fighting in caves more samey to fighting in every other part of the map. If caves have too many unpushable chokepoints, you can modify the map instead.

**Comment by Losenis** [comment]
>Not everything needs to have a direct counter.

If something doesn't have a counter there's no reason to not keep using it over and over and over, since it'd have better chances of working than something that does have a counter. It's why every single game marines make the attempt to form a gigantic murderball. It doesn't work out all the time not because it gets countered, but because marines are generally too stupid to realize it doesn't have a counter/don't feel like joining for one reason or another.

**Comment by Swagile** [comment]
marines have T-gas, plasma cutters, SG's, flamers, incendiary grenades, and heavy ordinance (SADAR) to push caves

half of these marines get for free round start and i never see used properly or at all

making CAS able to bust open caves, the only place where you don't get fucked by CAS, is just gunna make defense even shittier

im up for this to be test merged but let me warn you that CAS is already bullshit to deal with in the open, especially orbit 1 CAS (and the indirect buff to CAS via voucher buys from req), and it will only get worse if you can now hit caves. doesn't matter if the cave buster missles are expensive, marines will just go orbit 5; the slow drop of CAS doesn't matter if it blows apart all the defenses (resin walls, turrets, etc) that xenos have in caves

**Comment by David-Stormwell** [comment]
Time to go PO and blow up all silos solo

**Comment by Chaznoodles** [comment]
Caves are caves for a reason. Powercreep.

File: code/modules/shuttle/cas_plane.dm (hunk 3 above)
Comment (BraveMole) [reply code/modules/shuttle/cas_plane.dm]: You did !a==b when you wanted to do !(a==b) or (a!=b). The first one you did was only working because penetrate_cave was a boolean, it will not work in general

**Comment by BraveMole** [comment]
I think the main benefits of this is also not making PO a boring ass role in a lot of the map. You're just sitting on your ass 50% of the time. Making it way less powerfull in caves but still a part of the marine arsenal seems ok to me

**Comment by Al-1ce** [comment]
> The penetrator missile costs 200 CAS points and has a explosive strength of (2, 4 7)
> The missile can fire in caves.

200 points is pretty cheap for the ability to 1 shot silos assuming someone can just pod down and report a location. I doubt you considered this to begin with though. I'd like to hear your idea for a counter on currently unavoidable artillery since all the xenos can do now is mop up the mess and re-weed.

> Marines often have a hard time pushing caves even with multiple miners and full map control

I'll mention it again, but with full miners and absolute control there should be nothing stopping marines from pushing caves.
If you seriously are having trouble with top-tier gear consider using RRs or flamethrowers. I've seen it happen maybe once or twice where very, very well equiped marines somehow out-unga the xenos and die. It's often well deserved.

>Xenos can flip a marine victory to a xeno victory by spamming walls, turrets and other form of slow down defenses (all forms of defenses that could be countered by fire support and wait until they accumulate enough larva to counter attack.

I can't stress enough this isn't a mechanical fault and instead one by the marines at play. Use grenades.

>The basics of game design state that everything should have a counter. Within caves, spam has no counter.

No. I can name off a few things like CAS and OB which cannot be countered and only ignored or barely mitigated at best (jelly, for one with fire/OB), -With everything being supposed to have a counter as you've said, how do you expect xenos to counter this?

> The lack of fire support and wall spam makes caves too hard to push. This should help make that less of a problem.

This was, and is, intentional by design. If you've got 5 B-18s and multiple SG's and can't push caves thats a skill problem. Caves are not supposed to be easy to push, but your descriptions are either hyperbole or plainly unrealistic.

PR needs a serious revamp without being gamebreaking, but overall I can understand where you're coming from.

**Comment by JackTheJackhammer** [comment]
Should I make the cost 350 points?

File: code/modules/shuttle/cas_plane.dm (hunk 3 above)
Comment (JackTheJackhammer) [reply code/modules/shuttle/cas_plane.dm]: Ah, well I removed the ==TRUE so that should fix it

**Comment by usnpeepoo** [comment]
with the point buying from req, it'd be pretty scary for xenos trying to avoid cas in caves when all of a sudden, the only place you'd think would be safe from cas is now gone.

**Review by TheBimmer231 — APPROVED** [review]

File: code/game/objects/structures/dropship_ammo.dm [hunk 5]
```diff
@@ -231,6 +233,19 @@
 /obj/structure/ship_ammo/rocket/detonate_on(turf/impact, attackdir = NORTH)
 	qdel(src)
 
+/obj/structure/ship_ammo/rocket/cavebuster
+	name = "\improper GBU-1172 'Penetrator'"
+	desc = "The GBU-1172 'Penetrator' is a joint TGMC-NT Prototype. It is used to destroy and kill bunkers. It is designed to puncture the top steel plating of bunkers and detonate a mild, slightly radioactive charge into the plating, shattering it. Goes straight through mountains. Use at your own risk. NT-TGMC JOINT PROJECTS GROUP A is not liable for any Marine or PMC injuries when using this weapon. Use with Caution."
```
Comment (XSlayer300) [inline code/game/objects/structures/dropship_ammo.dm:238]:
```suggestion
	desc = "The GBU-1172 'Penetrator' is a joint TGMC-NT prototype missile. It is used to destroy bunkers as well as very fortified positions such as caves. It is designed to puncture the top steel plating of bunkers and detonate a mild, slightly radioactive charge into the plating, shattering it. Use at your own risk. NT-TGMC Join Project Group A is not liable for any TGMC personnel injured when using this weapon. "
```

Comment (JackTheJackhammer) [reply code/game/objects/structures/dropship_ammo.dm:238]: I cant seem to be able to change the change suggest at the top. So, I'm going to say it here. Joint is spelled wrong

**Comment by davidchan13** [comment]
> The basics of game design state that everything should have a counter. Within caves, spam has no counter.

The irony here is you are giving a role that has no counter a buff it doesn't need. CAS is 100% uncounterable by xenos. Once the flare is up even if acid is put on it right away CAS still has 60 seconds to drop their payload. Xenos don't have reliable ways to sneak aboard the ship to attack Condor pilot, even a good wraith can't just stowaway on the Alamo without being detected.

CAS doesn't need any buffs right now and can wipe entire hives when used properly, giving them the ability to hit inside caves, even limited amounts, is unnecessary.

**Comment by neohenry** [comment]
> 
> 
> > The basics of game design state that everything should have a counter. Within caves, spam has no counter.
> 
> The irony here is you are giving a role that has no counter a buff it doesn't need. CAS is 100% uncounterable by xenos. Once the flare is up even if acid is put on it right away CAS still has 60 seconds to drop their payload. Xenos don't have reliable ways to sneak aboard the ship to attack Condor pilot, even a good wraith can't just stowaway on the Alamo without being detected.
> 
> CAS doesn't need any buffs right now and can wipe entire hives when used properly, giving them the ability to hit inside caves, even limited amounts, is unnecessary.

CAS is hivemind that can actually hurt mobs instead of indirectly walling off retreat paths.

**Comment by BraveMole** [comment]
> > The basics of game design state that everything should have a counter. Within caves, spam has no counter.
> 
> The irony here is you are giving a role that has no counter a buff it doesn't need. CAS is 100% uncounterable by xenos. Once the flare is up even if acid is put on it right away CAS still has 60 seconds to drop their payload. Xenos don't have reliable ways to sneak aboard the ship to attack Condor pilot, even a good wraith can't just stowaway on the Alamo without being detected.
> 
> CAS doesn't need any buffs right now and can wipe entire hives when used properly, giving them the ability to hit inside caves, even limited amounts, is unnecessary.

Fair points. At the very least, marine not having cas in caves make the start of "unga rush silo" basicly a failiure almost every time. Why not giving cas utility in caves at least? like lights, small tanglefoot, that kind of stuff? cause cas pilot is an awfull role in map like lv where marines have a foot hold in caves

**Comment by Swagile** [comment]
> 
> 
> > > The basics of game design state that everything should have a counter. Within caves, spam has no counter.
> > 
> > 
> > The irony here is you are giving a role that has no counter a buff it doesn't need. CAS is 100% uncounterable by xenos. Once the flare is up even if acid is put on it right away CAS still has 60 seconds to drop their payload. Xenos don't have reliable ways to sneak aboard the ship to attack Condor pilot, even a good wraith can't just stowaway on the Alamo without being detected.
> > CAS doesn't need any buffs right now and can wipe entire hives when used properly, giving them the ability to hit inside caves, even limited amounts, is unnecessary.
> 
> Fair points. At the very least, marine not having cas in caves make the start of "unga rush silo" basicly a failiure almost every time. Why not giving cas utility in caves at least? like lights, small tanglefoot, that kind of stuff? cause cas pilot is an awfull role in map like lv where marines have a foot hold in caves

this would honestly be a better solution; giving CAS quasi mortar abilities (light HE, no incendiary, tangles, and light cave buster rounds)

by light HE i mean like mini rocket level at best

**Comment by Smudged** [comment]
> > > The basics of game design state that everything should have a counter. Within caves, spam has no counter.
> > 
> > 
> > The irony here is you are giving a role that has no counter a buff it doesn't need. CAS is 100% uncounterable by xenos. Once the flare is up even if acid is put on it right away CAS still has 60 seconds to drop their payload. Xenos don't have reliable ways to sneak aboard the ship to attack Condor pilot, even a good wraith can't just stowaway on the Alamo without being detected.
> > CAS doesn't need any buffs right now and can wipe entire hives when used properly, giving them the ability to hit inside caves, even limited amounts, is unnecessary.
> 
> Fair points. At the very least, marine not having cas in caves make the start of "unga rush silo" basicly a failiure almost every time. Why not giving cas utility in caves at least? like lights, small tanglefoot, that kind of stuff? cause cas pilot is an awfull role in map like lv where marines have a foot hold in caves

I think the biggest issues is just LV's map design forcing xeno's to defend in the only spot they can, in the caves, opening up more avenues for the invisible hand of CAS to block off routes, or make things generally a bad time is just not ideal in my eyes.

Simply a better choice to instead fix LV for Xeno's instead of making it even worse.

**Comment by Naaanii** [comment]
Get good and learn how to push in caves with the tools you actually have instead of requiring an air support crutch.

File: code/game/objects/structures/dropship_ammo.dm (hunk 5 above)
Comment (XSlayer300) [reply code/game/objects/structures/dropship_ammo.dm:238]:
```suggestion
	desc = "The GBU-1172 'Penetrator' is a joint TGMC-NT prototype missile. It is used to destroy bunkers as well as very fortified positions such as caves. It is designed to puncture the top steel plating of bunkers and detonate a mild, slightly radioactive charge into the plating, shattering it. Use at your own risk. NT-TGMC Joint Project Group A is not liable for any TGMC personnel injured when using this weapon. "
```

**Comment by TiviPlus** [comment]
Caves are impenetrable for a reason
